### ISSUE TITLE
npm install frida failed

### ISSUE THREAD

**Issue by prncoprs** [frida/frida-node#79]
**Describe the bug**
When I use npm to install frida, it failed.

**To Reproduce**
`npm install frida`

**Error Outputs**
```
➜  frida-node git:(main) ✗ npm install frida
npm ERR! code 1
npm ERR! path /Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/frida
npm ERR! command failed
npm ERR! command sh -c -- prebuild-install || node-gyp rebuild
npm ERR! prebuild-install warn install No prebuilt binaries found (target=19.1.0 runtime=node arch=arm64 libc= platform=darwin)
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.1.0
npm ERR! gyp info using node@19.1.0 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.10.8 found at "/opt/homebrew/opt/python@3.10/bin/python3.10"
npm ERR! gyp info spawn /opt/homebrew/opt/python@3.10/bin/python3.10
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/frida/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/admin/Library/Caches/node-gyp/19.1.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/admin/Library/Caches/node-gyp/19.1.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/admin/Library/Caches/node-gyp/19.1.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/frida',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp: binding.gyp not found (cwd: /Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/frida) while trying to load binding.gyp
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/node-gyp/lib/configure.js:284:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm ERR! gyp ERR! System Darwin 21.2.0
npm ERR! gyp ERR! command "/opt/homebrew/Cellar/node/19.1.0/bin/node" "/Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/.bin/node-gyp" "rebuild"
npm ERR! gyp ERR! cwd /Users/admin/Projects/SDKPrivacyGuidance/tools/frida-node/node_modules/frida
npm ERR! gyp ERR! node -v v19.1.0
npm ERR! gyp ERR! node-gyp -v v9.1.0
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/admin/.npm/_logs/2022-11-19T03_29_48_472Z-debug-0.log
```

**Desktop (please complete the following information):**
 - OS: Mac mini(m1, 2020),  MacOS 12.1
 - nodejs: v19.1.0
 - npm: 8.19.3

**Comment by simkoc** [comment]
I cannot help fixing the bug but I have found a work around. The issue seems to be with the nodejs version. Using [nvm](https://wiki.archlinux.org/title/Node.js#Alternate_installations) to install `nodejs@18.11.0` enabled me to install frida without a hitch.

**Comment by oleavr** [comment]
This is fixed in Frida 16.0.3, released the other day. The issue was that we didn't provide prebuilds for Node.js v19.

**Comment by liziwork** [comment]
How did you solve it?
